src/jit/decomposelongs.cpp

```diff
@@ -920,15 +925,24 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
 
     GenTree* tree   = use.Def();
     GenTree* gtLong = tree->gtGetOp1();
+    GenTree* oldShiftOp = tree->gtGetOp2();
+
```

Comment:
Shiftwidth op is a lclvar. Every time I tried to get the CNS_INT that the lcl var stored, I got compile errors. Also, in the case of shift being a constant, I don't think we actually need the store to lcl var, so maybe that should actually be moved down to the else case.